src/binheap.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""Create a binary heap."""
+from __future__ import unicode_literals
+
+
+class BinNode(object):
+    """Create a Node constructor to store data in a binary heap."""
+
+    def __init__(self, value, parent=None):
+        """Create an instance of a Node in a binary heap."""
+        self.value = value
+        self.parent = parent
+        self.child_a = None
+        self.child_b = None
+
+    def __gt__(self, other):
+        """Magic method for greater than."""
+        return self.value > other.value
+
+    def __lt__(self, other):
+        """Magic method for less than."""
+        return self.value < other.value
+
+
+class BinHeap(object):
+    """Create a Binary Heap constructor."""
+
+    def __init__(self, data_list=None):
+        """Create an instance of a Binary Heap in max mode."""
+        self.top = None
+        self.bin_heap_list = []
+        if data_list:
+            try:
+                data_list.sort()
+            except TypeError as message:
+                    raise TypeError('The input is of different types; cannot create a binary heap.' + message)
+            except AttributeError as message:
+                    raise AttributeError('The input is not sortable; cannot create a binary heap.' + str(message))
+
+            data_list.reverse()
```

Comment:
You can make your program faster by combining the sort and reverse actions. `list.sort()` and also the builtin function `sorted` both take a `reverse` keyword, which takes a boolean value.